src/Middleware/Rewrite/src/ApacheModRewrite/Tokenizer.cs

```diff
@@ -108,7 +108,19 @@ private static void RemoveQuotesAndEscapeCharacters(IList<string> tokens)
             {
                 var token = tokens[i];
                 var trimmed = token.Trim('\"');
-                tokens[i] = Regex.Unescape(trimmed);
+
+                if (trimmed.Contains("\\d") || trimmed.Contains("\\w") || trimmed.Contains("\\s"))
+                {
+                    tokens[i] = trimmed;
```

Comment:
Hrm, is it problematic that we're not calling Unescape at all if any one of these are present? What if there are other characters in the trimmed string that **do** need to be unescaped?

My concern is that this may need to be much more complicated and individually process each escape sequence.

Reply:
Why do we need to unescape some characters at all? Due to documentation of [Apache mod_rewrite](https://httpd.apache.org/docs/2.4/rewrite/intro.html), file with rules must contain [Perl Compatible Regular Expressions](http://pcre.org/), so rules should not be unescaped.

Reply:
@anurse Do you still think we need to be escaping here?